### PR TITLE
Azure CosmosDB module

### PR DESCRIPTION
**Pull request by tomconte**
I am proposing this CosmosDB module and example I wrote for using on a project I am working on, where we needed to assert certain Cosmos properties after deployment. I tried to follow the model of other Azure module contributions.

### CODE REVIEW

**Comment by rguthriemsft** [comment]
resolves #741

File: examples/azure/terraform-azure-cosmosdb-example/README.md
Comment (rguthriemsft) [inline examples/azure/terraform-azure-cosmosdb-example/README.md:5]: Can you follow the description format we used in VM example here https://github.com/gruntwork-io/terratest/blob/master/examples/azure/terraform-azure-vm-example/README.md

File: examples/azure/terraform-azure-cosmosdb-example/main.tf
Comment (rguthriemsft) [inline examples/azure/terraform-azure-cosmosdb-example/main.tf:31]: add a postfix variable to your variables.tf and use in naming similar to [this](https://github.com/gruntwork-io/terratest/blob/b13ada22754d4ec78aa0f3a838f9030071fee22b/examples/azure/terraform-azure-vm-example/main.tf#L30) to avoid collisions in name during CI.

File: examples/azure/terraform-azure-cosmosdb-example/main.tf
Comment (rguthriemsft) [inline examples/azure/terraform-azure-cosmosdb-example/main.tf:40]: probably need postfix here as well.  see comment above.

File: examples/azure/terraform-azure-cosmosdb-example/main.tf
Comment (rguthriemsft) [inline examples/azure/terraform-azure-cosmosdb-example/main.tf:43]: we want to use the smallest instance/sku sizes to reduce cost during CI as it runs on every pull request multiple times usually.  If this is smallest, that is fine.

File: examples/azure/terraform-azure-cosmosdb-example/variables.tf
Comment (rguthriemsft) [inline examples/azure/terraform-azure-cosmosdb-example/variables.tf:9]: add postfix variable.

File: modules/azure/cosmosdb.go
Comment (rguthriemsft) [inline modules/azure/cosmosdb.go:14]: Switch GetClient function to have sovereign cloud support.  doc for how to do this is [here](https://terratest.gruntwork.io/docs/community/contributing/#azure-sdk-client-factory).

File: modules/azure/cosmosdb.go
Comment (rguthriemsft) [inline modules/azure/cosmosdb.go:12]: In Terratest almost all methods support foo and fooE methods to allow for better error handling.  Pattern is documented [here](https://terratest.gruntwork.io/docs/testing-best-practices/error-handling/), please follow this pattern also.

File: examples/azure/terraform-azure-cosmosdb-example/main.tf
Comment (rguthriemsft) [inline examples/azure/terraform-azure-cosmosdb-example/main.tf:67]: consider making throughput a variable in variables.tf so that in your test you can assert the expected value you passed in.

File: test/azure/terraform_azure_cosmosdb_example_test.go
Comment (rguthriemsft) [inline test/azure/terraform_azure_cosmosdb_example_test.go:59]: see comment about about throughput value as a variable.

**Review by rguthriemsft — CHANGES_REQUESTED** [review]
@tomconte, ping me internally if you need help getting CI setup to run against your fork.  We documented the instructions here https://terratest.gruntwork.io/docs/community/contributing/#validate-the-pull-request-for-azure-platform

**Comment by tomconte** [comment]
[Microsoft CI Bot] TL;DR; success :thumbsup:

You can check the status of the CI Pipeline logs here ; https://github.com/tomconte/terratest/actions/runs/433914570

File: examples/azure/terraform-azure-cosmosdb-example/main.tf
Comment (tomconte) [reply examples/azure/terraform-azure-cosmosdb-example/main.tf]: Got it, done.

File: examples/azure/terraform-azure-cosmosdb-example/main.tf
Comment (tomconte) [reply examples/azure/terraform-azure-cosmosdb-example/main.tf]: Done.

File: examples/azure/terraform-azure-cosmosdb-example/variables.tf
Comment (tomconte) [reply examples/azure/terraform-azure-cosmosdb-example/variables.tf:9]: Done.

File: examples/azure/terraform-azure-cosmosdb-example/main.tf
Comment (tomconte) [reply examples/azure/terraform-azure-cosmosdb-example/main.tf:43]: Yes, the cost of Cosmos DB is based on RU/s (request units per second), the minimum being 400.

File: examples/azure/terraform-azure-cosmosdb-example/main.tf
Comment (tomconte) [reply examples/azure/terraform-azure-cosmosdb-example/main.tf]: Makes sense, done.

File: test/azure/terraform_azure_cosmosdb_example_test.go
Comment (tomconte) [reply test/azure/terraform_azure_cosmosdb_example_test.go]: Done.

File: modules/azure/cosmosdb.go
Comment (tomconte) [reply modules/azure/cosmosdb.go]: Done.

File: modules/azure/cosmosdb.go
Comment (tomconte) [reply modules/azure/cosmosdb.go]: Hey Richard, I implemented this, however the tests fail when running in my tenant (the MS one), for China and German clouds (see error below)... Do you know how to get around this? Or will this work in your CI environment?

```
***"error":"invalid_resource","error_description":"AADSTS500011: The resource principal named https://management.chinacloudapi.cn/ was not found in the tenant named 72f988bf-86f1-41af-91ab-2d7cd011db47. This can happen if the application has not been installed by the administrator of the tenant or consented to by any user in the tenant. You might have sent your authentication request to the wrong tenant.\r\nTrace ID: 355d2142-339d-4d8e-ac28-bbe39a2e2b00\r\nCorrelation ID: 84fdfe4e-c3b5-49ba-bbc8-87b7ea12ea03\r\nTimestamp: 2020-12-21 11:49:18Z","error_codes":[500011],"timestamp":"2020-12-21 11:49:18Z","trace_id":"355d2142-339d-4d8e-ac28-bbe39a2e2b00","correlation_id":"84fdfe4e-c3b5-49ba-bbc8-87b7ea12ea03","error_uri":"https://login.microsoftonline.com/error?code=500011"***
```

Here is my CI run: https://github.com/tomconte/terratest/runs/1588685618?check_suite_focus=true

Comment (tomconte) [reply modules/azure/cosmosdb.go]: Never mind, I found what was wrong. Should be OK now.

**Comment by tomconte** [comment]
[Microsoft CI Bot] TL;DR; success :thumbsup:

You can check the status of the CI Pipeline logs here ; https://github.com/tomconte/terratest/actions/runs/437863521

File: examples/azure/terraform-azure-cosmosdb-example/README.md
Comment (tomconte) [reply examples/azure/terraform-azure-cosmosdb-example/README.md]: I have updated the README to clarify the resources configured by the Terraform template.

File: modules/azure/cosmosdb.go
Comment (yorinasub17) [inline modules/azure/cosmosdb.go:147]: It looks like a lot of these comments need to be updated. E.g., this function is returning the DB throughput, not the database itself, right?

**Review by yorinasub17 — COMMENTED** [review]
Code implementation looks good to me! The documentation for the functions need another pass to make sure they are consistent with each function though. Once that is done, I can approve and merge this in!

File: modules/azure/cosmosdb.go
Comment (tomconte) [reply modules/azure/cosmosdb.go]: Thanks for your feedback! I have updated the comments.

**Review by yorinasub17 — APPROVED** [review]
Updates LGTM! I can merge this once I see the link to the successful build logs in the fork.

**Comment by tomconte** [comment]
[Microsoft CI Bot] TL;DR; success :thumbsup:

You can check the status of the CI Pipeline logs here ; https://github.com/tomconte/terratest/actions/runs/472130821

**Comment by yorinasub17** [comment]
@tomconte Can you double check the build? It looks like https://github.com/tomconte/terratest/actions/runs/472130821 refers to a different commit from the latest on this branch (the build is on 907c09f, not f804abb).

**Comment by tomconte** [comment]
[Microsoft CI Bot] TL;DR; success :thumbsup:

You can check the status of the CI Pipeline logs here ; https://github.com/tomconte/terratest/actions/runs/472470521

**Comment by tomconte** [comment]
@yorinasub17 I guess I selected the wrong branch when running the build 😅 Should be good now?

**Comment by yorinasub17** [comment]
Thank you! Build looks good to me, so will go ahead and merge this in!

**Pull request merged**
